File: app/src/main/java/info/meysam/coverto/data/OrderDao.java
package info.meysam.coverto.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import info.meysam.coverto.data.models.Order;

@Dao
public interface OrderDao {

    @Query("SELECT * FROM orders")
    List<Order> getAll();

    @Insert
    void insert(Order order);

    @Update
    void update(Order order);





}

File: app/src/main/java/info/meysam/coverto/data/OrderDao.kt
package info.meysam.coverto.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import info.meysam.coverto.data.models.Order

@Dao
interface OrderDao {


    @Insert
    fun insert(order: Order)

    @Update
    fun update(order: Order)


    @get:Query("SELECT * FROM orders")
    val all: ArrayList<Order?>?


    @Query("DELETE FROM orders WHERE id = :id")
    fun delete(id: Int)
}